src/approaches/interactive_learning_approach.py

```diff
@@ -21,86 +24,106 @@ class InteractiveLearningApproach(NSRTLearningApproach):
-            for s in traj.states:
-                ground_atoms = utils.all_possible_ground_atoms(
-                    s, self._predicates_to_learn)
-                for atom in ground_atoms:
-                    # Note: future score functions will use the state s
-                    score = score_atom(dataset, atom)
-                    # Ask about this atom if it is the best seen so far
-                    if score > best_score:
-                        if self._ask_teacher(s, atom):
-                            # Add this atom if it's a positive example
-                            traj = LowLevelTrajectory([s], [])
-                            dataset.append(traj, [{atom}])
-                            # Still need a way to use negative examples
-                        best_score = score
-            if i % CFG.interactive_relearn_every == 0:
-                self._relearn_predicates_and_nsrts(dataset,
-                                                   online_learning_cycle=i - 1)
+        # Save dataset, to be used for online interaction
+        self._dataset = dataset
+
+    def get_interaction_requests(self) -> List[InteractionRequest]:
+        # Sample a train task
+        train_task_idx = self._rng.choice(len(self._train_tasks))
+        init = self._train_tasks[train_task_idx].init
+        # Detect and filter out static predicates
+        static_preds = utils.get_static_preds(
+            self._nsrts, self._get_current_predicates())
+        preds = self._get_current_predicates() - static_preds
+        # Find acting policy for the request
+        task_list = glib_sample(init, preds, self._dataset)
+        assert task_list
+        task, act_policy = self._find_first_solvable(task_list)
+        assert task.init is init
+        def _query_policy(s: State) -> Optional[GroundAtomHoldsQuery]:
+            # Decide whether to ask about each possible atom
+            ground_atoms = utils.all_possible_ground_atoms(
+                s, self._predicates_to_learn)
+            for atom in ground_atoms:
+                score = score_atom(self._dataset, atom)
+                # Ask about this atom if it is the best seen so far
+                if score >= self._best_score:
+                    # TODO: the previous line is a > on master, changed to >=
+                    # because otherwise i wasn't seeing any new data get added
+                    # (due mostly to the next TODO)
+                    self._best_score = score
+                    # TODO: one difference from current master is that
+                    # we can now only ask one Query per state, whereas
+                    # before you could ask the teacher multiple things
+                    # about the same state. hopefully this is still okay...
+                    # otherwise we can change the InteractionRequest object's
+                    # query object to allow a set of Querys per timestep
```

Comment:
good point. I think we do want to allow the agent to ask about multiple atoms per state. we could do what you suggest here, or we could change the Query/Response for this project to be about sets of ground atoms instead of one. I don't have a preference.

Reply:
I decided to go with the latter: #381